lib/instance.js

```diff
@@ -897,22 +899,11 @@ Instance.prototype.restore = function(options) {
  * @return {Promise<this>}
  */
 Instance.prototype.increment = function(fields, options) {
-  var identifier = this.where()
+  var identifier = Utils.mapOptionFieldNames({ where: this.where() }, this.Model).where
```

Comment:
I know i'm just nitpicking now, but i imagine we might have to repeat this code more than these two times, perhaps we could provide a second `fields`|boolean argument to `where()` that will apply this code?